astropy/stats/src/fast_sigma_clip.c

```diff
@@ -100,17 +103,30 @@ static void _sigma_clip_fast(
     npy_intp is_mask = *steps++;
 
     double *buffer = NULL;
+    double *buffer2 = NULL;
 
+    // buffer is used to store the current values being sigma clipped
     buffer = (double *)PyArray_malloc(n_i * sizeof(double));
     if (buffer == NULL) {
         PyErr_NoMemory();
         return;
     }
+
+    // buffer2 is available to be used by compute_sigma_clipped_bounds
```

Comment:
Ideally, we only allocate this when actually needed. The data arrays can be quite large. I'd suggest setting it to NULL and then inside the loop check whether `use_mad` is set, and if so replace a NULL by a memory allocation. Obviously, in that case, the freeing also needs to check whether the buffer was allocated in the first place.

Reply:
I am running out of time for the feature freeze so can we get this in and the follow up can always go in later as it would not be an additional feature just an optimization of existing code?